### PR TITLE
fix: enable client parser to retain carriage return characters

### PR DESCRIPTION
**Pull request by kingflamez**
## What is the motivation for this pull request?

Bug fix

## What is the current behavior?

Addresses [html-dom-parser #420](https://github.com/remarkablemark/html-dom-parser/issues/420)
Addresses [html-react-parser #864](https://github.com/remarkablemark/html-react-parser/issues/864)

## What is the new behavior?

const text = "toto \r\n toto"

parse(text)

old output: "toto toto"
new output: "toto \r\n toto"

## Checklist:

- [x] [Conventional Commits](https://www.conventionalcommits.org/)
- [ ] Tests

### CODE REVIEW

File: src/client/domparser.ts [hunk 1]
```diff
@@ -109,13 +109,29 @@ if (template && template.content) {
   };
 }
 
+/**
+ * Escapes special characters before parsing.
+ *
+ * @param html - The HTML string.
+ * @returns - HTML string with escaped special characters.
+ */
+function escapeSpecialCharacters(html: string): string {
```
Comment (remarkablemark) [inline src/client/domparser.ts:118]: Can you add a test?

File: src/client/utilities.ts [hunk 2]
```diff
@@ -51,6 +51,16 @@ function formatTagName(tagName: string): string {
   return tagName;
 }
 
+/**
+ * Reverts escaped special characters back to actual characters.
+ *
+ * @param text - The text with escaped characters.
+ * @returns - Text with escaped characters reverted.
+ */
+function revertEscapedCharacters(text: string): string {
```
Comment (remarkablemark) [inline src/client/utilities.ts:60]: Can you add a test?

**Review by remarkablemark — COMMENTED** [review]
Looks good, left 2 comments

**Comment by kingflamez** [comment]
@remarkablemark added tests

**Review by remarkablemark — APPROVED** [review]
LGTM!

**Pull request merged**